be/src/exec/spill/spiller.hpp

```diff
@@ -0,0 +1,156 @@
+    auto captured_mem_table = std::move(_mem_table);
+    if (captured_mem_table == nullptr) {
+        return Status::OK();
+    }
+
+    RETURN_IF_ERROR(captured_mem_table->done());
+    _running_flush_tasks++;
+    // TODO: handle spill queue
+    auto task = [this, state, guard = guard, mem_table = std::move(captured_mem_table)]() {
+        SCOPED_TIMER(_metrics.flush_timer);
+        DCHECK_GT(_running_flush_tasks, 0);
+        DCHECK(has_pending_data());
+        guard.scoped_begin();
+        //
+        auto defer = DeferOp([&]() {
+            {
+                std::lock_guard guard(_mutex);
+                _mem_table_pool.emplace(std::move(mem_table));
+            }
+
+            _update_spilled_task_status(_decrease_running_flush_tasks());
+        });
+
+        auto status_task = [state, this, mem_table]() -> Status { return _run_flush_task(state, mem_table); };
+        _update_spilled_task_status(status_task());
+        guard.scoped_end();
+    };
+    // submit io task
+    RETURN_IF_ERROR(executor.submit(std::move(task)));
+    return Status::OK();
+}
+
+template <class TaskExecutor, class MemGuard>
+StatusOr<ChunkPtr> Spiller::restore(RuntimeState* state, TaskExecutor&& executor, MemGuard&& guard) {
+    SCOPED_TIMER(_metrics.restore_timer);
+    RETURN_IF_ERROR(_spilled_task_status);
+    ChunkPtr chunk;
+
+    if (_current_stream == nullptr) {
+        ASSIGN_OR_RETURN(_current_stream, _acquire_input_stream(state));
+    }
+
+    DCHECK(has_output_data());
+    // read chunk from buffer
+    ASSIGN_OR_RETURN(chunk, _current_stream->read(_spill_read_ctx));
+    TRACE_SPILL_LOG << "read rows:" << chunk->num_rows() << " cumulative:" << _restore_read_rows
+                    << ", spiller:" << this;
+    COUNTER_UPDATE(_metrics.restore_rows, chunk->num_rows());
+    _restore_read_rows += chunk->num_rows();
+
+    RETURN_IF_ERROR(trigger_restore(state, std::forward<TaskExecutor>(executor), std::forward<MemGuard>(guard)));
+
+    return chunk;
+}
+
+template <class TaskExecutor, class MemGuard>
+Status Spiller::trigger_restore(RuntimeState* state, TaskExecutor&& executor, MemGuard&& guard) {
+    if (_current_stream == nullptr) {
+        ASSIGN_OR_RETURN(_current_stream, _acquire_input_stream(state));
+    }
+
+    std::queue<SpillRestoreTaskPtr> capature_restore_tasks;
+    {
+        std::lock_guard guard(_mutex);
+        capature_restore_tasks = std::move(_restore_tasks);
+    }
+    // submit restore task
+    while (!capature_restore_tasks.empty()) {
+        DCHECK(capature_restore_tasks.front() != nullptr);
+        auto task = std::move(capature_restore_tasks.front());
+        RETURN_IF_ERROR(executor.submit([this, state, guard, task = std::move(task)]() {
+            _running_restore_tasks++;
+            guard.scoped_begin();
+
+            auto caller = [state, this, task]() -> Status {
+                SpillFormatContext spill_ctx;
+                RETURN_IF_ERROR(task->do_read(spill_ctx));
```

Comment:
Where are read chunks stored?

Reply:
see the detail implements in BufferedSpillReadTask